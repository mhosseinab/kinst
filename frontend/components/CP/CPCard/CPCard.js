import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Collapse, Card } from 'antd';
import cs from 'classnames';
import s from './CPCard.scss';

const { Panel } = Collapse;
const { Meta } = Card;

class CPCard extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    title: PropTypes.node,
    className: PropTypes.string,
    type: PropTypes.oneOf([
      'default',
      'success',
      'info',
      'danger',
      'warning',
      'inner',
    ]),
    more: PropTypes.node,
    close: PropTypes.bool,
    toggle: PropTypes.bool,
    noBorder: PropTypes.bool,
    cover: PropTypes.node,
    actions: PropTypes.array,
    metaAvatar: PropTypes.node,
    metaTitle: PropTypes.string,
    metaDesc: PropTypes.node,
  };

  static defaultProps = {
    children: '',
    title: '',
    className: null,
    type: 'default',
    more: null,
    close: false,
    toggle: false,
    noBorder: false,
    cover: '',
    actions: [],
    metaAvatar: null,
    metaTitle: '',
    metaDesc: '',
  };

  constructor(props) {
    super(props);
    this.state = {
      close: false,
    };
  }

  // handle close button functionality
  handleClose = () => {
    this.setState({ close: !this.state.close });
  };

  // render card extra parts (actions: more, toggle button, close button)
  renderCardActions = () => {
    const { title, close, more } = this.props;
    return (
      <span>
        <span className={s.title}>{title}</span>
        <span className={s.extra}>
          {more && <span className={s.more}>{more}</span>}
          {close && (
            <Icon
              type="close"
              className={s.closeIcon}
              onClick={this.handleClose}
            />
          )}
        </span>
      </span>
    );
  };

  render() {
    const {
      className,
      type,
      children,
      toggle,
      title,
      more,
      noBorder,
      metaAvatar,
      metaDesc,
      metaTitle,
      cover,
      actions,
    } = this.props;
    let style;

    switch (type) {
      case 'success': {
        style = s.success;
        break;
      }
      case 'danger': {
        style = s.danger;
        break;
      }
      case 'info': {
        style = s.info;
        break;
      }
      case 'warning': {
        style = s.warning;
        break;
      }
      default: {
        style = s.default;
        break;
      }
    }

    if (this.state.close) {
      return false;
    }

    return toggle ? (
      <Collapse
        defaultActiveKey={['1']}
        className={cs(s.root, className, style)}
      >
        <Panel header={this.renderCardActions()} key="1">
          {children}
        </Panel>
      </Collapse>
    ) : (
      <Card
        title={title}
        extra={more}
        className={cs(s.card, className, style)}
        bordered={!noBorder}
        cover={cover}
        actions={actions}
      >
        {metaAvatar || metaDesc || metaTitle ? (
          <Meta avatar={metaAvatar} title={metaTitle} description={metaDesc} />
        ) : (
          children
        )}
      </Card>
    );
  }
}

export default CPCard;
export const CPCardTest = CPCard;
