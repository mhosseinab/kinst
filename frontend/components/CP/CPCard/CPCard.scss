@import '../../variables.scss';

.card {
  padding: 20px;
  border-top: 2px solid;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 1px 6px rgba(0, 0, 0, 0.12);

  :global(.ant-card-meta-avatar) {
    padding-right: 0;
    padding-left: 16px;
    float: right;
  }
}

.root {
  border-bottom: 1px solid #d9d9d9 !important;
  margin-bottom: 15px;
  border-top-width: 2px;
  border-radius: 2px;

  :global(.ant-collapse-header:focus) {
    outline: 0 !important;
  }

  :global(.ant-collapse-item .ant-collapse-header) {
    background: #fff !important;
  }

  :global(.ant-collapse-header .arrow) {
    position: relative !important;
    float: left;
    left: 0 !important;
  }

  .extra {
    float: left;
  }

  .title {
    color: rgba(0, 0, 0, 0.85);
    font-weight: 400;
  }

  .more {
    margin-left: 9px;
  }

  .closeIcon {
    font-size: 16px !important;
    height: 32px !important;
    border-radius: 50% !important;
    cursor: pointer !important;
    width: 32px !important;
    text-align: center !important;
    margin-top: 7px !important;
    line-height: 32px !important;
    float: left;

    &:hover {
      background: #d9d9d9 !important;
    }
  }
}

.default {
  border-top-color: #d2d6de;
}

.success {
  border-top-color: $color-success;
}

.info {
  border-top-color: $color-info;
}

.warning {
  border-top-color: $color-warning;
}

.danger {
  border-top-color: $color-danger;
}

