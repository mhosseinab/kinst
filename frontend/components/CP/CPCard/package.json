{
  "name": "CPCard",
  "version": "0.0.0",
  "private": true,
  "main": "./CPCard.js"
}
