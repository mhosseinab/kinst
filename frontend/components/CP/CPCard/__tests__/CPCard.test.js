import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import CPCard from '../CPCard';

describe('CPCard Test', () => {
  it('CPCard title props test"', () => {
    const wrapper = shallow(<CPCard title="Events" />);

    expect(wrapper.prop('title')).toEqual('Events');
  });
});

describe('CPCard Snapshot Testing', () => {
  it('CPCard Snapshot"', () => {
    const component = renderer.create(<CPCard />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
