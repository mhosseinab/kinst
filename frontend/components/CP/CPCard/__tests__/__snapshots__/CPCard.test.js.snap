// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CPCard Snapshot Testing CPCard Snapshot" 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    
  </div>
</div>
`;
